go/vt/grpcclient/client_auth_static.go

```diff
@@ -29,6 +32,11 @@ var (
 	credsFile string // registered as --grpc_auth_static_client_creds in RegisterFlags
 	// StaticAuthClientCreds implements client interface to be able to WithPerRPCCredentials
 	_ credentials.PerRPCCredentials = (*StaticAuthClientCreds)(nil)
+
+	clientCredsErr  error
+	clientCredsLock sync.Mutex
```

Comment:
I thought about whether it is possible to do this lock-free, and I believe not. So we do need to introduce this mutex.
To keep it consistent with the rest of Vitess (and it may even be golang convention), it should be renamed though.

Reply:
Yes, I think the mutex is needed to avoid a race 👍